File: VS/lab3/Fuel.cpp
#include "Fuel.hpp"
#include "SupplyRequestDetail.hpp"

Fuel::Fuel(int id, std::string name, FuelType type, float quantity)
    : Resource(id, name, MeasureUnit::eLiter, m_quantity), m_type(type) {
}

bool Fuel::IsMatches(SupplyRequestDetail* detail) const {
    return detail->GetSupplyType() == SupplyType::Fuel && detail->GetFuelType() == m_type;
}

FuelType Fuel::GetType() const {
    return m_type;
}

File: lab6/Java/src/warehouse/resources/Fuel.java
package warehouse.resources;

import enums.FuelType;
import enums.MeasureUnit;
import enums.SupplyType;
import java.util.Random;
import request.SupplyRequestDetail;

public class Fuel extends Resource implements Cloneable {
    FuelType type;

    public Fuel(int id, String name, FuelType type, int count)
    {
        super(id, name, MeasureUnit.LITER, count);
        this.type = type;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.FUEL &&
               detail.getFuelType() == type;
    }

    public FuelType getType() {
        return type;
    }

    @Override
    public Fuel clone() throws CloneNotSupportedException{
        var clone = (Fuel)super.clone();
        clone.id = new Random().nextInt(0,9999);
        return clone;
    }
    
}
